src/libs/Navigation/NavigationRoot.js

```diff
@@ -1,115 +1,52 @@
 import PropTypes from 'prop-types';
-import {
-    getStateFromPath,
-    getPathFromState,
-    NavigationContainer,
-} from '@react-navigation/native';
-import {withOnyx} from 'react-native-onyx';
+import {getPathFromState, NavigationContainer} from '@react-navigation/native';
 import {navigationRef} from './Navigation';
 import linkingConfig from './linkingConfig';
 import AppNavigator from './AppNavigator';
-import getPathName from './getPathName';
-import ONYXKEYS from '../../ONYXKEYS';
 import ROUTES from '../../ROUTES';
-import styles from '../../styles/styles';
-import themeColors from '../../styles/themes/default';
 import {updateCurrentlyViewedReportID} from '../actions/Report';
 import {setCurrentURL} from '../actions/App';
+import FullScreenLoadingIndicator from '../../components/FullscreenLoadingIndicator';
 
 const propTypes = {
     // Whether the current user is logged in with an authToken
     authenticated: PropTypes.bool.isRequired,
-
-    // The current reportID that we are navigated to or should show in the ReportScreen
-    currentlyViewedReportID: PropTypes.string,
-};
-
-const defaultProps = {
-    currentlyViewedReportID: null,
 };
 
 class NavigationRoot extends Component {
     constructor(props) {
         super(props);
 
-        this.state = {
-            loading: true,
-            initialState: undefined,
-        };
+        this.handleStateChange = this.handleStateChange.bind(this);
     }
 
-    componentDidMount() {
-        Linking.getInitialURL()
-            .then((initialUrl) => {
-                // On web we should be able to parse this. It will be null on native for now until deep links are
-                // hooked up
-                const path = getPathName(initialUrl);
-                let initialState = getStateFromPath(path, linkingConfig.config);
-                setCurrentURL(path);
-
-                // If we are landing on something other than the report screen or site root then we MUST set the
-                // initial route to the currently viewed report so there some history to navigate back from
-                if (path !== `/${ROUTES.HOME}` && !path.includes(`/${ROUTES.REPORT}`)) {
-                    const homeRoute = {
-                        name: 'Home',
-                    };
-
-                    if (this.props.currentlyViewedReportID) {
-                        homeRoute.params = {
-                            screen: 'Report',
-                            params: {
-                                reportID: this.props.currentlyViewedReportID,
-                            },
-                        };
-                    }
-
-                    if (!initialState) {
-                        initialState = {
-                            routes: [],
-                        };
-                    }
+    /**
+     * Intercept state changes and perform different logic
+     * @param {NavigationState} state
+     */
+    handleStateChange(state) {
```

Comment:
I like that this was moved, but our style guide says to not use `handle` prefix and instead pick a name that describes what the method does e.g. `updateStoredRoutes()` or whatever you like.

Reply:
Yep, I know, just couldn't come up with a name.
I wasn't sure if this handler wouldn't grow in the future covering different state change cases so naming it after something now might not be relevant. E.g. the name I'm looking for is "something that decides what specific action should take place when a state changes" - it's just the decision maker and delegates to the specific action function

Reply:
Yeah it's just a style thing that we have for consistency. Since we are parsing the urls from the state we can make this `parseAndStoreRoute()` or something. If the responsibility grows we can change the name.